js_modules/dagit/packages/core/src/assets/usePartitionHealthData.tsx

```diff
@@ -286,16 +295,21 @@ export function keyCountInSelection(selections: PartitionDimensionSelectionRange
 //
 export function keyCountByStateInSelection(
   assetHealth: PartitionHealthData,
-  selections: PartitionDimensionSelection[],
+  _selections: PartitionDimensionSelection[],
 ) {
-  if (selections.length === 0) {
+  if (_selections.length === 0) {
     warnUnlessTest('[keyCountByStateInSelection] A selection must be provided for dimension 0.');
     return {
       [PartitionState.MISSING]: 0,
       [PartitionState.SUCCESS]: 0,
+      [PartitionState.FAILURE]: 0,
     };
   }
 
+  // Make sure that the provided selections are in the same order as the /underlying/
+  // range data, which may be reversed if the time series is the second axis.
+  const selections = assetHealth?.isRangeDataInverted ? [..._selections].reverse() : _selections;
```

Comment:
we should probably memoize this ?

Reply:
Ahh this is actually one of those hooks you were talking about where the hook returns an object with methods for interacting with the data. In this case selections is actually a param passed to a method it provides, so we can't memoize it here :-( I think it's only a 2 element array though so not too bad.